### PR TITLE
bpo-25810: Clarify eval() docs, it does not keywords (GH-15173)

### PR DESCRIPTION
**Pull request by smokephil**
[bpo-25810](https://bugs.python.org/issue25810): Clarify eval() docs, it does not keywords (GH-15173)

https://bugs.python.org/issue25810

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

File: Doc/library/functions.rst
Comment (aeros) [inline Doc/library/functions.rst:457]:
```suggestion
.. function:: eval(expression, globals=None, locals=None, /)
```
This clearly communications to the users that *expression*, *globals*, and *locals* are positional only parameters.

**Review by aeros — CHANGES_REQUESTED** [review]
Welcome and thanks for the contribution!

Based on the comment from R. David Murray:
>I think we are moving toward using (and documenting in the main docs) the / notation.  Especially since it shows up in the inspect module as well as the docstring help.

as well as the rest of the discussion on the issue, I think it would be more appropriate to use the / notation:

File: Doc/library/functions.rst
Comment (smokephil) [reply Doc/library/functions.rst:457]: I think it's easier to see with the square bracket notation that these are positional only arguments. It is also consistent with the exec documentation.

Comment (aeros) [reply Doc/library/functions.rst:457]: As mentioned in the issue discussion, the square bracket notation was previously used, and the `/` notation was specifically created to reduce the ambiguity. Without it, it's not entirely clear to users that they should not use keyword arguments for the *globals* and *locals* parameters. IMO, exec should be updated as well, unless a consensus was reached that differs from what was stated on the issue page.

We'll have wait on feedback from the core devs. In the meantime, I would recommend looking over [PEP 570](https://www.python.org/dev/peps/pep-0570/) if you're not already familiar with it.

Comment (gpshead) [reply Doc/library/functions.rst:457]: This PR is fine.  We use the traditional `[, []]` syntax all over the place in this file.  We do not use `, /)` anywhere in this .rst file.

`exec` is already documented using []s instead of wrongly implying it accepts named arguments.

Comment (aeros) [reply Doc/library/functions.rst:457]: @gpshead:
> This PR is fine. We use the traditional [, []] syntax all over the place in this file. We do not use , /) anywhere in this .rst file.

Is there any intention to move towards the usage of the PEP 570 syntax within this page at some point? I'm good with this PR using the old syntax if you're okay with it (since it's still an improvement from before either way), but my impression was that it was our intention to gradually move towards using it across the documentation. 

> exec is already documented using []s instead of wrongly implying it accepts named arguments.

Isn't the main purpose of the PEP 570 syntax to have a clear and consistent way to explicitly differentiate positional only parameters from named ones? ``param1=None, /, param2=None`` states that ``param1`` is an optional parameter that can't be passed a named argument and ``param2`` is an optional parameter that accepts positional or named arguments. That format is more clear than the brackets, which can get convoluted with the layers of nesting rather easily.

**Pull request merged**
